src/lib/operators/join_hash.cpp

```diff
@@ -322,33 +322,57 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
      */
 
     auto build_side_bloom_filter = BloomFilter{};
+    auto probe_side_bloom_filter = BloomFilter{};
 
-    Timer timer_materialization;
-    if (keep_nulls_build_column) {
-      materialized_build_column = materialize_input<BuildColumnType, HashedType, true>(
-          _build_input_table, _column_ids.first, histograms_build_column, _radix_bits, build_side_bloom_filter);
-    } else {
-      materialized_build_column = materialize_input<BuildColumnType, HashedType, false>(
-          _build_input_table, _column_ids.first, histograms_build_column, _radix_bits, build_side_bloom_filter);
-    }
-    _performance.set_step_runtime(OperatorSteps::BuildSideMaterializing, timer_materialization.lap());
+    const auto materialize_build_side = [&](const auto& input_bloom_filter) {
+      if (keep_nulls_build_column) {
+        materialized_build_column = materialize_input<BuildColumnType, HashedType, true>(
+            _build_input_table, _column_ids.first, histograms_build_column, _radix_bits, build_side_bloom_filter,
+            input_bloom_filter);
+      } else {
+        materialized_build_column = materialize_input<BuildColumnType, HashedType, false>(
+            _build_input_table, _column_ids.first, histograms_build_column, _radix_bits, build_side_bloom_filter,
+            input_bloom_filter);
+      }
+    };
 
     /**
      * 1.2. Materialize the larger probe partition. Use the bloom filter from the probe partition to skip rows that
      *       will not find a join partner.
      */
-    auto probe_side_bloom_filter = BloomFilter{};
+    const auto materialize_probe_side = [&](const auto& input_bloom_filter) {
+      if (keep_nulls_probe_column) {
+        materialized_probe_column = materialize_input<ProbeColumnType, HashedType, true>(
+            _probe_input_table, _column_ids.second, histograms_probe_column, _radix_bits, probe_side_bloom_filter,
+            input_bloom_filter);
+      } else {
+        materialized_probe_column = materialize_input<ProbeColumnType, HashedType, false>(
+            _probe_input_table, _column_ids.second, histograms_probe_column, _radix_bits, probe_side_bloom_filter,
+            input_bloom_filter);
+      }
+    };
 
-    if (keep_nulls_probe_column) {
-      materialized_probe_column = materialize_input<ProbeColumnType, HashedType, true>(
-          _probe_input_table, _column_ids.second, histograms_probe_column, _radix_bits, probe_side_bloom_filter,
-          build_side_bloom_filter);
+    Timer timer_materialization;
+    if (_build_input_table->row_count() < _probe_input_table->row_count()) {
+      materialize_build_side(ALL_TRUE_BLOOM_FILTER);
```

Comment:
If wouldn't have reviewed the PR with the `ALL_TRUE_BLOOM_FILTER` this would confuse me a lot. Maybe add a comment that states that there is still a bloom filter begin build, but the one passed as a parameter is the input bloom filter with which data will be pruned (so nothing being pruned here)?

Reply:
Agreed. Is this better?